nbclient/tests/test_util.py

```diff
@@ -75,3 +77,36 @@ async def test_run_hook_async():
     hook = MagicMock(return_value=some_async_function())
     await run_hook(hook)
     assert hook.call_count == 1
+
+
+def test_just_run_doesnt_leak_fds():
+    proc = psutil.Process()
+
+    async def async_sleep():
+        await asyncio.sleep(0.01)
+
+    # Warmup, just to make sure we're not failing on some initial fds being opened for the first time.
+    for _ in range(10):
+        just_run(async_sleep())
+    fds_count = proc.num_fds()
+
+    diff = []
+    for _ in range(10):
+        just_run(async_sleep())
+        diff.append(proc.num_fds() - fds_count)
+    assert diff == [0] * 10
+
+
+def test_just_run_clears_new_loop():
+    async def async_sleep():
+        await asyncio.sleep(0.1)
+
+    loop = asyncio.new_event_loop()
+    loop.stop = MagicMock(wraps=loop.stop)
+    loop.close = MagicMock(wraps=loop.close)
```

Comment:
Let's just ignore type checks on these.
```suggestion
    loop.stop = MagicMock(wraps=loop.stop)  # type: ignore
    loop.close = MagicMock(wraps=loop.close)  # type: ignore
```

Reply:
I think I might have a better solution for these, without ignoring type checks.

I can't run tests locally, though.
```
(venv) dev-envs-ad\mario@a-2zurb4h4bbny ➜  nbclient git:(mario/fix_fd_leak) ls
binder  CHANGELOG.md  CONTRIBUTING.md  docs  LICENSE  MANIFEST.in  nbclient  nbclient.egg-info  pyproject.toml  pytest.ini  README.md  RELEASING.md  requirements-dev.txt  requirements.txt  setup.cfg  setup.py  venv
(venv) dev-envs-ad\mario@a-2zurb4h4bbny ➜  nbclient git:(mario/fix_fd_leak) tox -e py38
ERROR: tox config file (either pyproject.toml, tox.ini, setup.cfg) not found
```

Reply:
We're not using `tox`, you should just run `pytest`.

Reply:
Ah I see, I was running `pytest`, but it was missing the type checking piece that failed. I think https://github.com/jupyter/nbclient/blob/main/CONTRIBUTING.md might need an update.

Re: type checking. Would it make sense to put it as pre-commit check?